includes/Modules/Analytics.php

```diff
@@ -937,115 +945,125 @@ protected function parse_data_response( $method, $datapoint, $response ) {
 					// TODO: Parse this response to a regular array.
 					break;
 				case 'accounts-properties-profiles':
-					$response = array(
-						// TODO: Parse this response to a regular array.
-						'accounts'   => $response->getItems(),
+					$data = array_merge(
+						array(
+							'existingAccountId'  => '',
+							'existingPropertyId' => '',
+						),
+						$data
+					);
+					/* @var \Google_Service_Analytics_Accounts $response listManagementAccounts response. */
+					$accounts            = (array) $response->getItems();
+					$account_ids         = array_map(
+						function ( \Google_Service_Analytics_Account $account ) {
+							return $account->getId();
+						},
+						$accounts
+					);
+					$properties_profiles = array(
 						'properties' => array(),
 						'profiles'   => array(),
 					);
 
-					$found_account_id = false;
-					$matched_property = false;
-					$existing_tag     = $this->_existing_tag_account;
-
-					$this->_existing_tag_account = null; // Set back to null.
-
-					if ( empty( $existing_tag ) ) {
+					if ( $data['existingAccountId'] && $data['existingPropertyId'] ) {
+						// If there is an existing tag, pass it through to ensure only the existing tag is matched.
+						$properties_profiles = $this->get_data(
+							'properties-profiles',
+							array(
+								'accountId'  => $data['existingAccountId'],
+								'propertyId' => $data['existingPropertyId'],
```

Comment:
I think this parameter should remain `existingPropertyId`, because it is a special scenario. There shouldn't be a datapoint with the main purpose to return a list of properties for an account to which you pass a property ID, because after all its purpose is to return all properties found.
To clarify with an example, I consider this similar to as if the `wp/v2/posts` API route (returning multiple posts) supported a `postId` parameter. This feels confusing. Naming it `existingPropertyId` makes it more obvious it's some kind of special circumstance.